Userland/Libraries/LibWeb/IntersectionObserver/IntersectionObserver.cpp

```diff
@@ -50,7 +50,7 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<IntersectionObserver>> IntersectionObserver
 IntersectionObserver::IntersectionObserver(JS::Realm& realm, JS::GCPtr<WebIDL::CallbackType> callback, Optional<Variant<JS::Handle<DOM::Element>, JS::Handle<DOM::Document>>> const& root, Vector<double>&& thresholds)
     : PlatformObject(realm)
     , m_callback(callback)
-    , m_root(root)
+    , m_root(root.has_value() ? root->visit([](auto& value) -> JS::GCPtr<DOM::Node> { return *value; }) : nullptr)
```

Comment:
would it be better to move this logic to an if/else in the body of the constructor? it's a bit inscrutable inline